### PR TITLE
web based term mapping

### PR DESCRIPTION
**Pull request by turbomam**
can now map any set of enums to any target ontology from bioportal
examples in Makefile
no live pull of tables yet (limited access Google sheets? Felix Postgres?)
but Makefile does include conversion of a manually obtained TSV to YAML

### CODE REVIEW

**Comment by turbomam** [comment]
@cmungall Is that failure from something I did? I haven't been working with dates/times.

PS, I haven't written any test of my own for the BioPortal mapping yet.

**Comment by turbomam** [comment]
I reprioritized OLS searches over BioPortal. I think this is ready to merge into master now, but the checks are failing at this unrelated code:

```
  coverage erase
  pytest --cov=kgx --cov-report xml:coverage.xml tests/*
  coverage combine --append || true
  coverage report
  coverage xml
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.7.10/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.7.10/x64/lib
Coverage.py warning: Module kgx was never imported. (module-not-imported)
```

File: linkml_model_enrichment/bioportal-enum-annotation.py [hunk 1]
```diff
@@ -8,17 +10,42 @@
...
+
+# go_prefixes_url = 'https://raw.githubusercontent.com/prefixcommons/biocontext/master/registry/go_context.jsonld'
+# r = requests.get(go_prefixes_url)
+#
+# mapping = CurieUtil.parseContext(r.json())
+# curie = CurieUtil(mapping)
+
+# WRITES OUTPUT TO STDOUT
+
+# 2021-04-14
```
Comment (cmungall) [inline linkml_model_enrichment/bioportal-enum-annotation.py:30]: minor: some of this belongs in git commit messages

in general no need to retain commented out code, git takes care of history

in general most docs should go in docstrings so they are visible at IDE/RTD level

File: linkml_model_enrichment/bioportal-enum-annotation.py [hunk 2]
```diff
@@ -89,54 +117,63 @@ def replace(string, substitutions):
 
 
 @click.command()
```
Comment (cmungall) [inline linkml_model_enrichment/bioportal-enum-annotation.py:119]: we may want to think about a yaml config file in the future

(the config file would be described using linkml)

File: linkml_model_enrichment/get_felix_tsv.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
```
Comment (cmungall) [inline linkml_model_enrichment/get_felix_tsv.py:1]: move to synbio repo

**Comment by wdduncan** [comment]
You may want to consider creating a pypi package for this, and attach command-line utils to the package.  I have examples of this in [nmdc-schema](https://github.com/microbiomedata/nmdc-schema).

File: Makefile [hunk 4]
```diff
@@ -1,4 +1,12 @@
+#!/usr/bin/env python
```
Comment (wdduncan) [inline Makefile:1]: I assume this will only run in python 3 ... if so you should use:
```
#!/usr/bin/env python3
```

File: Makefile [hunk 5]
```diff
@@ -1,4 +1,12 @@
...
+
 export PYTHONPATH=.
+.PHONY: data/felix_modifications.tsv
+.PHONY: clean
```
Comment (wdduncan) [inline Makefile:5]: You can have multiple `.PHONY` targets. I.e.:
```
.PHONY: data/felix_modifications.tsv clean
```

File: linkml_model_enrichment/get_felix_tsv.py (hunk 3 above)
Comment (turbomam) [reply linkml_model_enrichment/get_felix_tsv.py]: haven't moved yet, but have removed synbio mentions from this repo

**Pull request merged**